### PR TITLE
Fix a few concurrency bugs in Citus support

### PR DESCRIPTION
**Pull request by CyberDem0n**
- the `_in_fligh` attribute is accessed from multiple threads and must be protected with mutex when it is changed
- allow adding tasks for `_in_fligh.group` from the `sync_pg_dist_node()` method when timeout is reached. Not doing so might result is indefinite transaction if REST API request from worker node failed.

### CODE REVIEW

File: patroni/postgresql/citus.py [hunk 1]
```diff
@@ -293,10 +302,16 @@ def _add_task(self, task: PgDistNode) -> bool:
...
+            # The task.timeout is None is an indicator that it was scheduled from the sync_pg_dist_node().
+            # We don't want to override already existing task created from REST API.
+            # In case if there is already _in_flight we don't want to add a task until timeout is reached.
+            # The last one is important to protect from the case when REST API request failed.
+            # But, effectively we only need protection for the first few seconds until the member
+            # key was updated with the new state/role values after demotion has started.
+            if task.timeout is None\
+                and (i is not None and self._tasks[i].timeout is not None
+                     or self._in_flight and self._in_flight.group == task.group
+                     and self._in_flight.deadline > time.time()):
```
Comment (JelteF) [inline patroni/postgresql/citus.py:314]: This condition is pretty long and thus hard to parse for most humans. For these kinds of cases I prefer to split them up. Either in separate if statements, that simply do the same early return. Or I group them in names, e.g. 
```python
waiting_for_timeout = i is not None and self._tasks[i].timeout is not None
```

**Comment by hughcapet** [comment]
👍

**Comment by CyberDem0n** [comment]
:+1:
